components/page/main/chat-message.tsx

```
@@ -71,79 +88,881 @@ const darkTheme = {
+  const uniqueId = useMemo(() => `streaming-${Math.random().toString(36).substring(2, 9)}`, []);
+  
+  return (
+    <div className="relative streaming-markdown-container urgent-streaming-styles" 
+      ref={containerRef} 
+      data-id={uniqueId}>
+      <div className="prose-content streaming-content custom-streaming-markdown">
+        {useDirectHtml ? (
+          <div 
+            className="urgent-streaming-styles" 
+            data-id={uniqueId}
+            dangerouslySetInnerHTML={{ __html: createMarkdownHtml }}
```

Biome (1.9.4)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)